src/Conversion/ONNXToTOSA/Math/Gemm.cpp

```diff
@@ -0,0 +1,244 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===---------------- Gemm.cpp - Gemm Op ----------------------------------===//
+//
+// Copyright (c) 2022 Advanced Micro Devices, Inc.
+//
+// =============================================================================
+//
+// This file lowers ONNX Gemm operator to TOSA dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Dialect/Tosa/IR/TosaOps.h"
+#include "src/Conversion/ONNXToTOSA/DialectBuilder.hpp"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSACommon.hpp"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSALegalizeUtils.hpp"
+#include <cstdint>
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+namespace {
+
+class ONNXGemmOpLoweringToTOSA : public OpConversionPattern<ONNXGemmOp> {
+public:
+  using OpConversionPattern<ONNXGemmOp>::OpConversionPattern;
+
+  LogicalResult matchAndRewrite(ONNXGemmOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    TosaBuilder tosaBuilder(rewriter, op->getLoc());
+    // If legal, create a FullyConnected operator instead
+    if (rewriteToTosaFC(op, adaptor, rewriter, tosaBuilder)) {
+      return success();
+    }
```

Comment:
Forgotten if I've nit-picked on this before - does onnx-mlir not follow https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements?

Reply:
We recommend it.

Reply:
I removed the braces according to the LLVM Coding Standards.